### DISCUSSION TITLE
CIDR (IPv4) expansion as default behavior?

### DISCUSSION THREAD

**Discussion by funilrys** [funilrys/PyFunceble#231] [General]
Since https://github.com/funilrys/PyFunceble/commit/6834481e31c0bda3ef112c05191e616d83f5323f (which was pushed by accident🙃⁉️) pyfunceble is capable of expanding and testing CIDR formatted IPv4 addresses.

As of now, it is deactivated by default, as I'm not sure that everyone wants to suddenly get this behavior as default. 
Indeed, right now, when this option is not activated all valid IPv4 ranges (as a whole) are flagged as `ACTIVE` as part of the `SPECIAL` rules.

Therefore, I need more input from the end-users!

What do you think of this: Should we expand IPv4 ranges by default?

Feedback highly wanted!

cc: @mitchellkrogza @ZeroDot1 @spirillen @DandelionSprout

**Comment by spirillen** [comment]
As I mentioned as comment to the https://github.com/funilrys/PyFunceble/commit/6834481e31c0bda3ef112c05191e616d83f5323f commit and @funilrys comment in #150 that all IP in a CIDR notation was tested, And I believe they should be, I'll vote for it to be enabled by default.

The alternative is, you might end up with dead CIDR or inaccurate CIDR(ip) ranges. The other thing I can think of is speaking for it to be enabled (`True`) by default is, PyFunceble is designed to validate testing of any given records, not guessing, but testing for knowing.

**Reply by funilrys** [reply]
I agree, but I have the feeling that activating it by default will create some noise...

**Reply by spirillen** [reply]
> I agree, but I have the feeling that activating it by default will create some noise...

Yep, more replies would be welcome :wink:

**Reply by DandelionSprout** [reply]
I lack any actual opinions on this, for anyone wondering.

**Reply by spirillen** [reply]
> I agree, but I have the feeling that activating it by default will create some noise...

I have set the `cidr_expand: true` in [this action](https://github.com/dns-test/rpz.mypdns.cloud/actions) So let's see if anything comes up that could generate any "confusing" noise...

**Reply by funilrys** [reply]
Any feedback? I'm not into adding this as default but feedback gives me some idea of what is going on ...
